text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,626 @@
+A CloudFront distribution fronts this application, ensuring it is served from
+edge locations close to any customer, and binding together the static assets
+that compose the application and the data objects maintained by the back-end
+pipeline.
+
+1. The landing page asynchronously loads the `latest.json` object to render the
+   "latest" tiles, containing the short-list of packages that changed most
+   recently.
+
+1. The search feature asynchronously loads the `catalog.json` object, and
+   performs client-side filtering to prepare search results.
+
+1. Package detail pages fetch the relevant `assembly.json` object from the
+   assembly store, and renders the detail page using information contained
+   therein.
+
+   - If a different language (than JavaScript) is selected, the
+     `assembly-${lang}.json` object is fetched instead, providing the relevant
+     documentation elements. In the event that object is not available yet, a
+     message will be displayed instead, instructing the user to try again later.
+
+   URL                                                       | Description
+   ----------------------------------------------------------|----------------------------------------------------------
+   `<root>/packages[/@<scope>]/<name>`                       | Redirects to the latest known version of `[@scope/]name`
+   `<root>/packages[/@<scope>]/<name>/v/<version>`           | The overview of `[@scope/]name` at `version`
+   `<root>/packages[/@<scope>]/<name>/v/<version>/docs`      | The documentation (`README.md`) of `[@scope/]name` at `version`
+   `<root>/packages[/@<scope>]/<name>/v/<version>/docs/<FQN>`| The documentation for type `FQN` in `[@scope/]name` at `version`
+
+   - When a `<version>` parameter is only a prefix of a SemVer version number,
+     the URL redirects to the latest known version in that line (e.g: `/v/1`
+     would redirect to `/v/1.2.3`, and `/v/1.1` would redirect to `/v/1.1.9`)
+   - A query parameter (e.g: `?lang=go`) may be added to anchor a documentation
+     page (under the `/docs` scope) on a particular language.
+
+[react]: https://reactjs.org
+
+The details of URLs to be configrued in CloudFront is available in [appendix
+"CloudFront URLs"](#cloudfront-urls).
+
+#### Website Analytics
+
+An analytics solutions will be designed to enable monitoring the success and
+use-cases of the platform. A stream of URLs hit by users will be collected,
+including search terms used by users. That data will be completely anonymous,
+and an opaque identifier will be used to enable approximating "unique visitor"
+counts, etc. No cross-site tracking will be performed.
+
+If IP addresses are fed into the analyitics service, these will not be
+persisted, but may instead be mapped to a coarse-grained location (e.g:
+country).
+
+Legal review will be necessary to determine if the dataset implies compliance
+requirements for GDPR, CCPA, and similar consumer privacy regulations.
+
+#### Construct Packaging
```

Comment:
Yes - this is largely for illustration purposes, and to give a sense of the general direction. This isn't meant to be prescriptive at this stage (mostly it can serve as a way to identify minimal requirements).